File: DesignPatternsExercises/ruby/state_design_pattern/exercise/card_ejected_state.rb
require_relative 'atm_machine_states'

class CardEjectedState < AtmMachineStates
  def __________(8)__________
    puts 'No card in ATM Machine slot, unable to eject the card...'
    false
  end

  def insert_card?
    puts 'Card is inserted...'
    true
  end

  def withdraw?
    puts 'No card in ATM Machine slot, unable to withdraw...'
    false
  end
end

File: DesignPatternsExercises/java/StateDesignPattern/Exercise/CardEjectedState.java
public class CardEjectedState implements AtmMachineStates {
    @Override
    public boolean ejectCard() {
        System.out.println("No card in ATM Machine slot, unable to eject the card...");
        return false;
    }

    @Override
    public boolean insertCard() {
        System.out.println("Card is inserted...");
        return true;
    }

    @Override
    public boolean withdraw() {
        System.out.println("No card in ATM Machine slot, unable to withdraw...");
        return false;
    }
}
